### DISCUSSION TITLE
Find time-to-event of specific probability

### DISCUSSION THREAD

**Discussion by jacobkreider** [sebp/scikit-survival#239] [Q&A] [answered]
I'm trying to find the time-to-event based on the specific probability threshold. So, if I set a baseline of 75%, how would I get the step function returned via predict_survival_function to return that value?

**Comment by sebp** [comment, marked as answer]
You would need to invert the survival function. The exact details depend on how the survival function is constructed. For the Kaplan-Meier estimator, I explained it in https://github.com/sebp/scikit-survival/issues/15#issuecomment-884404114-permalink, for CoxPHSurvivalAnalysis, you would need to invert the [baseline survival function](https://scikit-survival.readthedocs.io/en/latest/api/generated/sksurv.linear_model.CoxPHSurvivalAnalysis.html#sksurv.linear_model.CoxPHSurvivalAnalysis.baseline_survival_) `S_0(t)`, because `S(t|x) = S_0(t)^{exp(x^T beta)}` you get `S_0(t) = exp[ log S(t|x) / exp(x^T beta) ]`, which leads to `t = S_0^{-1} ( exp[ log S(t|x) / exp(x^T beta) ] )`. Other formulations are possible too, e.g. by writing the survival function in terms of the cumulative hazard function first.

**Reply by jacobkreider** [reply]
Thanks for the quick reply!
